baseplate/config.py

```diff
@@ -335,30 +334,125 @@ def __init__(self):
         self.__dict__ = self
 
 
-def _parse_config_section(config, spec, root):
-    parsed = ConfigNamespace()
-    for key, parser_or_spec in spec.items():
-        assert "." not in key, "dots are not allowed in keys"
+class Parser(object):
+    """Base for config parsers."""
 
-        if root:
-            key_path = "%s.%s" % (root, key)
-        else:
-            key_path = key
-
-        if callable(parser_or_spec):
-            parser = parser_or_spec
-            raw_value = config.get(key_path, "")
-
-            try:
-                parsed[key] = parser(raw_value)
-            except Exception as e:
-                raise ConfigurationError(key, e)
-        elif isinstance(parser_or_spec, dict):
-            subspec = parser_or_spec
-            parsed[key] = _parse_config_section(config, subspec, root=key_path)
+    @classmethod
+    def from_spec(self, spec):
```

Comment:
If you do want to keep it a classmethod, could you rename `self` to `cls`?  Also, should the subclasses be calling `self.from_spec` rather than `Parser.from_spec`?